src/states/node.rs

```diff
@@ -1457,10 +1457,10 @@ impl Node {
             }
         }
 
-        let peers_needing_tunnel = self.peer_mgr.peers_needing_tunnel();
-        if !peers_needing_tunnel.is_empty() &&
-           self.peer_mgr.potential_tunnel_nodes().contains(&(*public_id.name(), *peer_id)) {
-            for dst_id in peers_needing_tunnel {
+        for (dst_id, peer_name) in self.peer_mgr.peers_needing_tunnel() {
+            if self.peer_mgr
+                .potential_tunnel_nodes(&peer_name)
```

Comment:
But instead of computing the set of all potential tunnel nodes, we could just:
```rust
(self.peer_mgr.routing_table().our_section().contains(&peer_name) ||
self.peer_mgr.routing_table().get_section(public_id.name()).map_or(false, |section| section.contains(&peer_name)))
&& self.peer_mgr.get_state_by_name(public_id.name()).map_or(false, PeerState::can_tunnel_for)
```
(Or, probably cleaner, move _that_ check into the peer manager.)